dependencies.yaml

```diff
@@ -62,8 +62,10 @@ dependencies:
     common:
       - output_types: [conda, requirements]
         packages:
+          - pre-commit
           - clang-tools==11.1.0
           - flake8
+          - GitPython==3.1.30
```

Comment:
I wouldn't put a hard pin here unless it's really necessary
```suggestion
          - GitPython>=3.1.30
```

Reply:
I think it is generally a bad idea to make our scripts vulnerable to arbitrary upstream changes in this way, but I understand  that this might be a separate discussion. I will apply your suggestion, but do we have a general consensus or guidance on this?

Reply:
Agreed, it's a separate discussion but for now removing the hard pin is probably best. As far as a general consensus, I would say that in most previous discussions we have generally come to agree with the ideas expressed in articles like [1](https://bernat.tech/posts/version-numbers/) and [2](https://iscinumpy.dev/post/bound-version-constraints/).